src/algebra/monoid_algebra/basic.lean

```diff
@@ -542,14 +542,32 @@ lemma single_one_comm [comm_semiring k] [mul_one_class G] (r : k) (f : monoid_al
 by { ext, rw [single_one_mul_apply, mul_single_one_apply, mul_comm] }
 
 /-- `finsupp.single 1` as a `ring_hom` -/
-@[simps] def single_one_ring_hom [semiring k] [monoid G] : k →+* monoid_algebra k G :=
+@[simps] def single_one_ring_hom [semiring k] [mul_one_class G] : k →+* monoid_algebra k G :=
 { map_one' := rfl,
   map_mul' := λ x y, by rw [single_add_hom, single_mul_single, one_mul],
   ..finsupp.single_add_hom 1}
 
+/--  A monoid homomorphism `f : G →* H` extends to a monoid homomorphism `G →* monoid_algebra k H`,
+mapping `n : G` to `finsupp.single (f x) 1`. -/
+def single_one_right_hom (k : Type*) {H : Type*} [semiring k] [mul_one_class G] [mul_one_class H]
+  (f : G →* H) : G →* monoid_algebra k H :=
+by refine ⟨λ x, single (f x) 1, _, _⟩; simp ; refl
+
+@[simp] lemma single_one_right_hom_apply (k : Type*) {H : Type*} [semiring k] [mul_one_class G]
+  [mul_one_class H] (f : G →* H) (n : G) :
+  single_one_right_hom k f n = single (f n) 1 := rfl
+
+/--  A multiplicative homomorphism `f : G →* H` between two monoids induces a `k`-algebra
+homomorphism `monoid_hom_lift f : monoid_algebra k G →+* monoid_algebra k H`. -/
+def monoid_hom_lift (k : Type*) {H : Type*} [semiring k] [monoid G] [monoid H]
```

Comment:
I think this should be called `map` not `lift`

Reply:
Name changed.

I also changed another name that stayed with its preliminary `full_lift`.